commands/new_function.go

```diff
@@ -55,21 +56,21 @@ func validateFunctionName(functionName string) error {
 	// Regex for RFC-1123 validation:
 	// 	k8s.io/kubernetes/pkg/util/validation/validation.go
 	var validDNS = regexp.MustCompile(`^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`)
-	matched := validDNS.MatchString(functionName)
-	if matched {
+	if matched := validDNS.MatchString(functionName); matched {
```

Comment:
We tend to advise against refactoring alongside another change, it can get distracting for review, but here's my take. Maybe the opposite order would improve the flow:

```
if matched := validDNS.MatchString(functionName); !matched {
    return fmt.Errorf(`function name can only contain a-z, 0-9 and dashes`)
}

return nil
```